modules/http/http.c

```diff
@@ -261,6 +263,37 @@ http_dd_set_ssl_version(LogDriver *d, const gchar *value)
   return TRUE;
 }
 
+void
+http_dd_set_accept_encoding(LogDriver *d, const gchar *encoding)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  if(strcmp(encoding, CURL_COMPRESSION_LITERAL_ALL) == 0) self->accept_encoding = "";
+  else self->accept_encoding = g_strdup(encoding);
+}
+
+void
+http_dd_set_message_compression(LogDriver *d, const gchar *encoding)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  gboolean _encoding_valid = FALSE;
+  _encoding_valid = http_dd_check_curl_compression(encoding);
+  g_assert(_encoding_valid);
+
+  if(http_dd_curl_compression_string_match(encoding, CURL_COMPRESSION_UNCOMPRESSED))
+    {
+      self->message_compression = CURL_COMPRESSION_UNCOMPRESSED;
+      return;
+    }
+  else if(http_dd_curl_compression_string_match(encoding, CURL_COMPRESSION_GZIP))
+    self->message_compression = CURL_COMPRESSION_GZIP;
+  else if(http_dd_curl_compression_string_match(encoding, CURL_COMPRESSION_DEFLATE))
+    self->message_compression = CURL_COMPRESSION_DEFLATE;
+  gchar *buffer = g_strdup_printf("Content-Encoding: %s", curl_compression_types[self->message_compression]);
+  self->headers= g_list_append(self->headers,  buffer);
```

Comment:
This should not be appended in the setter, as technically the user could set multiple compression in the config, where the last should be activated only. I think we could move this to the `init()` fn.

Reply:
Done